FerramAerospaceResearch/FARAeroComponents/FARAeroSection.cs

```diff
@@ -418,40 +507,26 @@ public void ClearAeroSection()
                 forceVector *= data.dragFactor;
                 torqueVector *= data.dragFactor;
 
-                applyLocalForce(data, forceVector, torqueVector, data.centroidPartSpace);
+                forceContext.ApplyForce(data, forceVector, torqueVector);
             }
         }
         
+        private SimulatedForceContext simContext;
         public void PredictionCalculateAeroForces(float atmDensity, float machNumber, float reynoldsPerUnitLength, float pseudoKnudsenNumber, float skinFrictionDrag, Vector3 vel, ferram4.FARCenterQuery center)
         {
-            CalculateAeroForces(atmDensity, machNumber, reynoldsPerUnitLength, pseudoKnudsenNumber, skinFrictionDrag,
-                pd => pd.aeroModule.part.partTransform.InverseTransformVector(vel),
-                (pd, forceVector, torqueVector, pcentroid) => {
-                    var localVel = pd.aeroModule.part.partTransform.InverseTransformVector(vel);
-                    var tmp = 0.0005 * Vector3.SqrMagnitude(localVel);
-                    var dynamicPressurekPa = tmp * atmDensity;
-                    var dragFactor = dynamicPressurekPa*Mathf.Max(PhysicsGlobals.DragCurvePseudoReynolds.Evaluate(atmDensity * Vector3.Magnitude(localVel)), 1.0f);
-                    var liftFactor = dynamicPressurekPa;
-
-                    var localVelNorm = Vector3.Normalize(localVel);
-                    Vector3 localForceTemp = Vector3.Dot(localVelNorm, forceVector) * localVelNorm;
-                    var partLocalForce = (localForceTemp * (float)dragFactor + (forceVector - localForceTemp) * (float)liftFactor);
-                    forceVector = pd.aeroModule.part.transform.TransformDirection(partLocalForce);
-                    torqueVector = pd.aeroModule.part.transform.TransformDirection(torqueVector * (float)dynamicPressurekPa);
-                    if (!float.IsNaN(forceVector.x) && !float.IsNaN(torqueVector.x))
-                    {
-                        Vector3 centroid = pd.aeroModule.part.transform.TransformPoint(pd.centroidPartSpace - pd.aeroModule.part.CoMOffset);
-                        center.AddForce(centroid, forceVector);
-                        center.AddTorque(torqueVector);
-                    }
-                });
+            if (simContext == null)
+                simContext = new SimulatedForceContext(vel, center, atmDensity);
+            else
+                simContext.UpdateSimulationContext(vel, center, atmDensity);
```

Comment:
https://github.com/dkavolis/Ferram-Aerospace-Research/blob/3459b30b70fda45e9573da643b0bcfffc3e427c2/FerramAerospaceResearch/FARAeroComponents/FARAeroSection.cs#L517-L520
https://github.com/dkavolis/Ferram-Aerospace-Research/blob/3459b30b70fda45e9573da643b0bcfffc3e427c2/FerramAerospaceResearch/FARAeroComponents/FARAeroSection.cs#L527-L528

Same as with `FARCenterQuery`, initialize in ctor and skip null checks